### ISSUE TITLE
 [AChecker] uploading file problem

### ISSUE THREAD

**Issue by lamu84** [inclusive-design/AChecker#74]
Hi,
 I tried to run both local and on line,  this file:(attached 087016 )
(I have this problems only with some files, and with others don't, all files in the same directory)
[087016.zip](https://github.com/inclusive-design/AChecker/files/1793201/087016.zip)

Local version AChecker 1.3
Xampp Php 5.3.0 
Browser Mozilla
Local AChecker returns me error.(like cannoct connect to url)
On line version runs correctly.
The local AChecker error could be related to the file or something else?
Could you please try to run in local this file and tell me if it runs correctly?
Thanks.

**Comment by cindyli** [comment]
Hi @lamu84, I tried the given html file with my local AChecker thru the tab "html file upload", the validation went thru and returned 48 known problems and 328 potential problems.

**Comment by lamu84** [comment]
did you do some improvement or tuning to your achecker build? because i still have this problem uploading the file. 
I have the same error pasting code html instead of uploading. 
Thanks again.

**Comment by cindyli** [comment]
No, I didn't. The investigation might happen at the next development cycle.

**Comment by lamu84** [comment]
May i ask you what kind of hardware (CPU, RAM) and softfware (OS, web platform for example xampp ) do you use to do your test? thanks.
is it possible that could be an HW limit ?

**Comment by cindyli** [comment]
@lamu84, my local AChecker runs on MacOS 10.13.3, MAMP environment with PHP 5.6.3 and MySQL 5.6.38, firefox 58.0.2 and chrome 64.

If your local achecker reports "cannot connect to url", open url in a separate browser to make sure it's accessible. If the problem persists, check PHP error log to see if there's any errors or warnings. If nothing looks wrong, debug at [this line](https://github.com/inclusive-design/AChecker/blob/master/checker/index.php#L149) where the url is fetched to find out what's wrong with `file_get_contents()`.

The validation does relate to the CPU processing power. It's a known issue that when the validated content contains deep nested html tags, the validation may not be able to finish before the timeout limit reaches. In this case, AChecker usually returns 500 or 502 "internal server error". You should be able to find a couple of existing bug reports related to this issue.
